### PR TITLE
Stabilize riak job

### PR DESCRIPTION
**Pull request by michalwski**
Proposed changes include:
* start `mod_keystore` and `mod_auth_token` only for tests
* stable riak job on travis

### CODE REVIEW

File: apps/ejabberd/src/mod_keystore.erl [hunk 1]
```diff
@@ -114,11 +114,18 @@ create_keystore_ets() ->
...
-            keystore = ets:new(keystore, [named_table, public,
-                                          {read_concurrency, true}]),
+            BaseOpts = [named_table, public,
+                        {read_concurrency, true}],
+            Opts = maybe_add_heir(whereis(ejabberd_sup), self(), BaseOpts),
+            keystore = ets:new(keystore, Opts),
             ok
     end.
 
+maybe_add_heir(EjdSupPid, _Self, BaseOpts) when is_pid(EjdSupPid) ->
```
Comment (ludwikbukowski) [inline apps/ejabberd/src/mod_keystore.erl]: what is a purpose of passing the variable `_Self` if it's not used?

Comment (michalwski) [inline apps/ejabberd/src/mod_keystore.erl]: Good catch, it's a leftover.

Comment (michalwski) [inline apps/ejabberd/src/mod_keystore.erl]: It was more than leftover, the `_Self` has to be used. I fixed this.

File: apps/ejabberd/src/mod_keystore.erl [hunk 2]
```diff
@@ -121,7 +121,11 @@ create_keystore_ets() ->
             ok
     end.
 
-maybe_add_heir(EjdSupPid, _Self, BaseOpts) when is_pid(EjdSupPid) ->
+%% In tests or when module is started in run-time, we need to set heir to the
+%% ETS table, otherwise it will be destroy when the creator's process finishes.
+%% When started normally during node start up, self() =:= EjdSupPid and there
+%% is no need for setting heir
+maybe_add_heir(EjdSupPid, EjdSupPid, BaseOpts) ->
     [{heir, EjdSupPid, testing} | BaseOpts];
```
Comment (ludwikbukowski) [inline apps/ejabberd/src/mod_keystore.erl]: I'm a bit confused. When calling `maybe_add_heir` function with the same `EjdSupPid` as a `_Self` pid we should avoid setting the heir (as it's clearly explained in comment). Didn't you mean:
```
maybe_add_heir(EjdSupPid, EjdSupPid, BaseOpts) when is_pid(EjdSupPid) ->
     BaseOpts;
maybe_add_heir(EjdSupPid, _Self, BaseOpts) when is_pid(EjdSupPid) ->
      [{heir, EjdSupPid, testing} | BaseOpts];
maybe_add_heir(_, _, BaseOpts) ->
      BaseOpts.
```

Comment (michalwski) [inline apps/ejabberd/src/mod_keystore.erl]: You are 100% right. Too many tasks in the same time, too less cores in my brain... I'll fix that. Thanks for careful review!

**Comment by bartekgorny** [comment]
Looks ok to me.

**Pull request merged**